### PR TITLE
Add OSCC-CODING.md file to describe the project coding standard

### PR DESCRIPTION
**Pull request by ltbj**
This PR introduces a style guide and coding standard for OSCC. 

All relevant information for a potential contributor to the OSCC project is contained within the `CONTRIBUTING.md` file. The bulk of the coding standard was pulled from the MISRA-C 2012 standard.

### CODE REVIEW

**Comment by nathanaschbacher** [comment]
Last I saw @petebrink had a mostly complete CONTRIBUTING.md file that incorprated this document, so that it didn't have to be a thing standing off by itself.  I'm inclined to want to have that pushed and considered for merging instead of this PR as-is.

**Comment by ltbj** [comment]
@petebrink nothing seems to be missing at first glance. However I would like to see a change to the top level bullets to make them numbers, so it's easy to call out specific violations with a section and reference number for the offender.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -25,22 +25,107 @@ Below is a sample of how additional vehicle directories should be created.
 ## Getting Started
 
 * Using the repo's Issues section, create a new issue and:
-  * Clearly describe the issue including steps to reproduce when it is a bug.
-* Fork the repository on GitHub
+  * Clearly describe the issue including steps to reproduce when it is a bug
+  * Branch the repository on GitHub
```
Comment (snewt) [inline CONTRIBUTING.md]: Branch? Seems like fork was the right word

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -4,15 +4,15 @@ There are many ways to contribute to the OSCC project; support for additional ve
 CAN frames can be better detailed, PCBs can be optimized
 or code could be refactored and improved.
 
-The directory structure of the project is created in such a way that adding support for additional vehicles should be simple and intuitive. 
+The directory structure of the project is created in such a way that adding support for additional vehicles should be simple and intuitive.
 All vehicle specific directories should be recreated for each additional vehicle. The naming convention for is <make>_<model>_<generational identifier>
```
Comment (annietupek) [inline CONTRIBUTING.md]: 'the naming convention for is' should be 'the naming convention for it is' or 'the naming convention is'

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -25,22 +25,535 @@ Below is a sample of how additional vehicle directories should be created.
...
+* Provide specific examples to demonstrate the steps
+    * Include snippets in the examples as Markdown code blocks ( using the \`\`\` format )
+* Describe the current behavior and how this enhancement improves or changes it
+* Explain why this enhancement is useful to most users
+* Specify which version of OSCC this change applies to
+    * The versions can be found in the Releases tab on GitHub
+
+### Pull Request Labels
+* Fill in the required template
+* Follow the style guides
```
Comment (annietupek) [inline CONTRIBUTING.md]: 'style guides' do we have multiple style guides, or should this be singular instead of plural?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -25,22 +25,535 @@ Below is a sample of how additional vehicle directories should be created.
...
+
+1. Implemented behavior must be documented and understood
+1. Comply with the C99 definition of the standard
+1. All source files must compile without errors
+1. Run-time failures must be limited
+	1. All arithmetic operations must be checked (e.g. divide by zero)
+	1. Pointer arithmetic must be checked
+	1. Array bounds must be checked
+	1. Check validity of parameters either before or inside a function call
+	1. Check pointer validity before dereference
```
Comment (annietupek) [inline CONTRIBUTING.md]: For consistent voice, number 4 & 5 could be reworded to 'Validity of parameters must be checked either before... and Pointer validity must be checked before dereference'

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -25,22 +25,535 @@ Below is a sample of how additional vehicle directories should be created.
...
+
+	- e.g. do not create a variable called “float”
+
+1. Standard library usage is acceptable
+
+### 18. Resources
+
+1. Free all allocated memory
+
+1. Do not open files for read and write at the same file, file access is read-only or write only
```
Comment (annietupek) [inline CONTRIBUTING.md]: '...for read and write at the same file...' did you mean 'at the same time?'

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -25,22 +25,535 @@ Below is a sample of how additional vehicle directories should be created.
...
+
+1. Logical comparisons should not contain persistent side effects such as assignments
+
+### 11. Control Statements
+
+1. Always use spaces instead of tabs
+	1. Use 4 space as the standard indent
+
+1. Include a space after an opening parenthesis, brace or bracket and a space before the closing match.
+	1. Parentheses: `( to_b || not_to_b )`
```
Comment (annietupek) [inline CONTRIBUTING.md]: I think our C-style guide has us putting parenthesis around each conditional in a multiple conditional check.  I didn't see anything about multiple-conditionals in this style-guide, though.  Is it something we want to add?

**Pull request merged**